# SPDX-License-Identifier: Apache-2.0
--- a/include/sys/video_controls.h
+++ b/include/sys/video_controls.h
@@ -58,8 +58,8 @@
 #define V4L2_CID_GAMMA                USER_CID(7)    /* Gamma value adjustment */
 #define V4L2_CID_GAMMA_CURVE          USER_CID(8)    /* Gamma curve adjustment */
 #define V4L2_CID_EXPOSURE             USER_CID(9)    /* Exposure value */
-#define V4L2_CID_AUTOGAIN             USER_CID(10)   /* Auto gain */
-#define V4L2_CID_GAIN                 USER_CID(11)   /* Gain value */
+#define V4L2_CID_AUTOGAIN             USER_CID(19)   /* Auto gain */
+#define V4L2_CID_GAIN                 USER_CID(20)   /* Gain value */
 
 /* Mirror horizontally(VIDEO) */
 
--- a/include/nuttx/video/imgsensor.h
+++ b/include/nuttx/video/imgsensor.h
@@ -46,6 +46,8 @@
 #define IMGSENSOR_ID_GAMMA                V4L2_CID_GAMMA
 #define IMGSENSOR_ID_GAMMA_CURVE          V4L2_CID_GAMMA_CURVE
 #define IMGSENSOR_ID_EXPOSURE             V4L2_CID_EXPOSURE
+#define IMGSENSOR_ID_AUTOGAIN             V4L2_CID_AUTOGAIN
+#define IMGSENSOR_ID_GAIN                 V4L2_CID_GAIN
 #define IMGSENSOR_ID_HFLIP_VIDEO          V4L2_CID_HFLIP
 #define IMGSENSOR_ID_VFLIP_VIDEO          V4L2_CID_VFLIP
 #define IMGSENSOR_ID_HFLIP_STILL          V4L2_CID_HFLIP_STILL
--- a/drivers/video/v4l2_cap.c
+++ b/drivers/video/v4l2_cap.c
@@ -407,6 +407,14 @@
     },
 
     {
+      IMGSENSOR_ID_AUTOGAIN,             "Automatic gain"
+    },
+
+    {
+      IMGSENSOR_ID_GAIN,                 "Gain value"
+    },
+
+    {
       IMGSENSOR_ID_HFLIP_VIDEO,          "Mirror horizontally(VIDEO)"
     },
 
