--- a/drivers/contactless/mfrc522.c
+++ b/drivers/contactless/mfrc522.c
@@ -356,7 +356,7 @@
   /* Wait for CRC completion or 200ms time-out */
 
   clock_systime_timespec(&tstart);
-  tstart.tv_nsec += 200000;
+  tstart.tv_nsec += 200000000;
   if (tstart.tv_nsec >= 1000 * 1000 * 1000)
     {
       tstart.tv_sec++;
@@ -377,7 +377,8 @@
 
       clock_systime_timespec(&tend);
 
-      if ((tend.tv_sec > tstart.tv_sec) && (tend.tv_nsec > tstart.tv_nsec))
+      if (tend.tv_sec > tstart.tv_sec ||
+          (tend.tv_sec == tstart.tv_sec && tend.tv_nsec >= tstart.tv_nsec))
         {
           return -ETIMEDOUT;
         }
@@ -455,7 +456,7 @@
    */
 
   clock_systime_timespec(&tstart);
-  tstart.tv_nsec += 200000;
+  tstart.tv_nsec += 200000000;
   if (tstart.tv_nsec >= 1000 * 1000 * 1000)
     {
       tstart.tv_sec++;
@@ -497,8 +498,8 @@
 
       clock_systime_timespec(&tend);
 
-      if ((tend.tv_sec > tstart.tv_sec) &&
-          (tend.tv_nsec > tstart.tv_nsec))
+      if (tend.tv_sec > tstart.tv_sec ||
+          (tend.tv_sec == tstart.tv_sec && tend.tv_nsec >= tstart.tv_nsec))
         {
           return -ETIMEDOUT;
         }
